### ISSUE TITLE
Large request payloads cause the server to error and the request to time out

### ISSUE THREAD

**Issue by aaron-novarca** [serverpod/serverpod#1994]
**Describe the bug**
When a request payload exceeds a certain size (I estimate this to be around 530,000 bytes), the server errors and the request is never completed and times out.

**To Reproduce**
I have attached an example of a request object that causes this bug to occur. If you send a similar object to any endpoint you should be able reproduce it.
[request-obj.json](https://github.com/serverpod/serverpod/files/14510226/request-obj.json)

**Screenshots**
![Screenshot 2024-03-06 at 12 56 10](https://github.com/serverpod/serverpod/assets/128369434/ab47f477-3c65-4408-8bb1-910626e5fd25)

**Serverpod versions**
 -  _CLI Version:_ 1.2.4
 - _Library version:_ 1.2.4

**Platform information**
Platform independent, has occurred in the past both with the server running locally on Linux and macOS, as well as when the server is hosted on AWS.

**Comment by vlidholt** [comment]
This needs to be documented. By default, the maximum request size is set to 512KB as a safeguard for spamming the server. You can up this limit by setting the `maxRequestSize` key in your server's config.

E.g., this will double the request size to 1MB in your development environment:
`config/development.yaml`

```yaml
maxRequestSize: 1048576

apiServer:
  port: 8080
  publicHost: localhost
  publicPort: 8080
  publicScheme: http

# Configuration for the Insights server.
insightsServer:
  port: 8081
  publicHost: localhost
  publicPort: 8081
  publicScheme: http

# Configuration for the web server.
webServer:
  port: 8082
  publicHost: localhost
  publicPort: 8082
  publicScheme: http

# This is the database setup for your server.
database:
  host: localhost
  port: 8090
  name: chat
  user: postgres

# This is the setup for Redis.
redis:
  enabled: false
  host: localhost
  port: 8091
```

**Comment by vlidholt** [comment]
Will be fixed in the 2.1 docs.
